File: src/forza/telemetry/VehicleData.java
package forza.telemetry;

public class VehicleData {
    private ForzaTelemetryApi forzaApi;
    public VehicleData(ForzaTelemetryApi forzaApi) {
        this.forzaApi = forzaApi;
    }
    public String getCarName() {
        return forzaApi.getCarName();
    }

    public String getCarClass() {
        return forzaApi.getCarClass();
    }

    public Integer getPerformanceIndex() {
        return forzaApi.getPerformanceIndex();
    }

    public String getDrivetrain() {
        return forzaApi.getDrivetrain();
    }

    public Integer getNumOfCylinders() {
        return forzaApi.getNumOfCylinders();
    }

    public String getCarType() {
        return forzaApi.getCarType();
    }

    public Integer getOrdinal() {
        return forzaApi.getOrdinal();
    }
}

File: kotlin_lib/VehicleData.kt
package com.fasetto.fragmentjava.kotlin_lib

class VehicleData(private val forzaApi: ForzaTelemetryApi) {
    val carName: String
        get() = forzaApi.getCarName()
    val carClass: String
        get() = forzaApi.getCarClass()
    val performanceIndex: Int
        get() = forzaApi.getPerformanceIndex()
    val drivetrain: String
        get() = forzaApi.getDrivetrain()
    val numOfCylinders: Int
        get() = forzaApi.numOfCylinders
    val carType: String
        get() = forzaApi.getCarType()
    val ordinal: Int
        get() = forzaApi.getOrdinal()
}